keanu-project/src/main/java/io/improbable/keanu/algorithms/graphtraversal/BreadthFirstSearch.java

```diff
@@ -0,0 +1,52 @@
+package io.improbable.keanu.algorithms.graphtraversal;
+
+import io.improbable.keanu.vertices.Vertex;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class BreadthFirstSearch {
+
+    public static boolean bfs(Collection<Vertex> vertices,
```

Comment:
Looks like bfs does more than just traverse and visit. It seems to allow some true/false check. This is fine but we need to do better to describe this in the name of the class and method.